rust/lance-encoding/src/data.rs

```diff
@@ -25,8 +29,18 @@ use snafu::{location, Location};
 
 use lance_core::{Error, Result};
 
-use crate::buffer::LanceBuffer;
+use crate::{buffer::LanceBuffer, statistics::Stat};
 
+/// `Encoding` enum serves as a encoding registeration center.
+/// All the encodings added to Lance should register here, and
+/// these encodings can be dynamically selected during encoding,
+/// users can also specify the particular encoding they want to use in the field metadata.
+#[derive(Debug)]
+pub enum Encoding {
+    Bitpack,
+    Fsst,
+    FixedSizeBinary,
+}
```

Comment:
we can put this enum into a dedicated file `encodings.rs` later, each registered encoding also need to supply the `DataBlock` type they can work with, optionally, each registered encoding can also provide a cost function(to compute the compression ratio, and even more things)

Reply:
I am thinking about making all encodings a trait object(inspired by datafusion's buildin function trait object)